cookbooks/cdo-varnish/libraries/helpers.rb

```diff
@@ -45,10 +45,12 @@ def normalize_paths(paths)
       (is_extension && path.match('/'))
       raise ArgumentError.new("Invalid path: #{path}")
     end
+    single_path = !path.match('\*')
```

Comment:
It should be the opposite- the `single_path` variable matches against any path that _does not_ contain `*`, and the appended `QUERY_REGEX` matches against an (optional) query string followed by an end-of-string anchor, so it needs to be applied to the two path-expression types that don't have a wildcard at the end. The end-of-string match makes sure that extensions ("`*.jpg`") and single-path expressions (`"/foo"`) don't match against expressions like `"not_image.jpgxyz"` or `"/foobar"`.

As for updating the comment I could update the name and description of `QUERY_REGEX` to make it more clear its main purpose is to match the end of the path, unless you have other suggestions to make this more clear.